### PR TITLE
fix(core): carousel, fix console error, arrows

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue

closes https://github.com/SAP/fundamental-ngx/issues/5969

## Description

 fixed carousel issues

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [x] RTL/LTR - proper rendering and labeling
-   [x] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [n/a] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [n/a] check for console logs (warnings, errors)
-   [n/a] API boundary values
-   [n/a] different combinations of components - free style
-   [n/a] change the API values during testing

3. Documentation and Example validations

-   [n/a] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

**Review by genereg1 — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/carousel/examples/carousel-dynamic-items-example.component.html [hunk 1]
```diff
@@ -216,7 +216,7 @@ <h3 fd-card-subtitle>Card Seven Subtitle</h3>
         </fd-carousel-item>
 
         <fd-carousel-item>
-            <div style="width: 300px; padding: 1rem">
+            <div style="width: 300px; padding: 1rem; box-sizing: border-box">
```
Comment (N1XUS) [inline apps/docs/src/app/core/component-docs/carousel/examples/carousel-dynamic-items-example.component.html:219]: Can this be moved to a separate class of the example component?

File: libs/core/src/lib/carousel/carousel.component.ts [hunk 2]
```diff
@@ -350,12 +356,8 @@ export class CarouselComponent implements OnInit, AfterContentInit, AfterViewIni
 
     /** @hidden Subscribe to carousel service events */
     private _subscribeServiceEvents(): void {
-        this._carouselService.activeChange
-            .pipe(takeUntil(this._onDestroy$))
-            .subscribe((event) => this._onSlideSwipe(event));
-        this._carouselService.dragStateChange
-            .pipe(takeUntil(this._onDestroy$))
-            .subscribe((event) => this._onSlideDrag(event));
+        this._carouselService.activeChange.subscribe((event) => this._onSlideSwipe(event));
```
Comment (N1XUS) [inline libs/core/src/lib/carousel/carousel.component.ts:359]: How's this subscription being destroyed?

**Review by N1XUS — APPROVED** [review]

**Review by dmitry-stepanenko — APPROVED** [review]

File: libs/core/src/lib/carousel/carousel.component.html [hunk 3]
```diff
@@ -36,13 +36,13 @@
...
     <div class="fd-carousel__page-indicator-container">
-        <ng-container *ngIf="navigatorInPageIndicator">
+        <ng-container *ngIf="navigatorInPageIndicator && pageIndicatorsCountArray.length > 0">
             <ng-container *ngTemplateOutlet="buttonLeft"></ng-container>
         </ng-container>
 
         <ng-container *ngTemplateOutlet="pageIndicators"></ng-container>
 
-        <ng-container *ngIf="navigatorInPageIndicator">
+        <ng-container *ngIf="navigatorInPageIndicator && pageIndicatorsCountArray.length > 0">
```
Comment (aberikashvili) [inline libs/core/src/lib/carousel/carousel.component.html:45]: I would extract those kind of things to a separate method

File: libs/core/src/lib/carousel/carousel.service.ts [hunk 4]
```diff
@@ -99,6 +95,9 @@ export class CarouselService implements OnDestroy {
     ngOnDestroy(): void {
         this._onDestroy$.next();
         this._onDestroy$.complete();
+
+        this.activeChange.complete();
+        this.dragStateChange.complete();
```
Comment (aberikashvili) [inline libs/core/src/lib/carousel/carousel.service.ts:100]: I see, these are not new properties, but still I'd recommend to add `$` sign as a suffix

File: libs/core/src/lib/carousel/carousel.component.ts (hunk 2 above)
Comment (Betrozov) [reply libs/core/src/lib/carousel/carousel.component.ts:359]: here https://github.com/SAP/fundamental-ngx/pull/7529/files/254b75e477b072ba95c8a9c34483892e36d44c39#diff-6a760f57d72b55490dd32c11c868561f957d368ee7ff27b7401b9ffb83e633f6R99

**Review by aberikashvili — APPROVED** [review]
LGTM (just few recommendations)

**Pull request merged**